pytorch_lightning/trainer/data_loading.py

```diff
@@ -153,6 +161,10 @@ def prepare_dataloader(self, dataloader: Any, shuffle: bool, mode: Optional[Runn
             sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
             dataloader = self._update_dataloader(dataloader, sampler, mode=mode)
 
+        if cycle_iterator is not None:
```

Comment:
shouldn't the dataloader variable be set to the cycle_iterator reference and this if-statement be an instance check instead?

Reply:
Hey @awaelchli, not sure to fully get your suggestion.

The cycle_iterator isn't defined in some case. How would you like to implement this ?